atoma-sui/src/handlers.rs

```diff
@@ -1,85 +1,492 @@
+///
+/// * `Result<()>` - Ok(()) if the event was processed successfully, or an error if something went wrong.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// * The event data cannot be deserialized into a `StackCreatedEvent`.
+/// * The database operation to insert the new stack fails.
+///
+/// # Behavior
+///
+/// The function only inserts the new stack into the database if the selected node's ID
+/// is present in the `node_small_ids` slice. This ensures that only stacks relevant to
+/// the current node(s) are processed and stored.
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_stack_created_event(
+    value: Value,
+    db: &SqlitePool,
+    node_small_ids: &[u64],
+) -> Result<()> {
+    trace!("Processing stack created event");
+    let stack_event: StackCreatedEvent = serde_json::from_value(value)?;
+    let node_small_id = stack_event.selected_node_id.inner;
+    if node_small_ids.contains(&node_small_id) {
+        trace!("Stack selected current node, with id {node_small_id}, inserting new stack");
+        let state_manager = StateManager::new(db.clone());
+        let stack = stack_event.into();
+        state_manager.insert_new_stack(stack).await?;
+    }
+    Ok(())
+}
+
+/// Handles a stack try settle event.
+///
+/// This function processes a stack try settle event by parsing the event data,
+/// converting it into a stack settlement ticket, and inserting it into the database.
+///
+/// # Arguments
+///
+/// * `value` - A `serde_json::Value` containing the serialized stack try settle event data.
+/// * `db` - A reference to the `SqlitePool` for database operations.
+///
+/// # Returns
+///
+/// * `Result<()>` - Ok(()) if the event was processed successfully, or an error if something went wrong.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// * The event data cannot be deserialized into a `StackTrySettleEvent`.
+/// * The database operation to insert the new stack settlement ticket fails.
+///
+/// # Behavior
+///
+/// The function performs the following steps:
+/// 1. Deserializes the input `value` into a `StackTrySettleEvent`.
+/// 2. Converts the event into a stack settlement ticket.
+/// 3. Creates a new `StateManager` instance.
+/// 4. Inserts the new stack settlement ticket into the database using the `StateManager`.
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_stack_try_settle_event(value: Value, db: &SqlitePool) -> Result<()> {
+    trace!("Processing stack try settle event");
+    let stack_settlement_ticket_event: StackTrySettleEvent = serde_json::from_value(value)?;
+    let stack_settlement_ticket = stack_settlement_ticket_event.into();
+    let state_manager = StateManager::new(db.clone());
+    state_manager
+        .insert_new_stack_settlement_ticket(stack_settlement_ticket)
+        .await?;
+    Ok(())
+}
+
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_new_stack_settlement_attestation_event(
```

Comment:
this function as well

Reply:
Ďakujem, že ste si všimli chýbajúce časti.